hermes/probe/create_file_test.go

```diff
@@ -1 +1,201 @@
+	want, wantErr := "", "The file ID provided 51 wasn't in the required range [1,50]"
+	got, err := testUnits[0].FileName()
+	gotErr:=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{51, 12}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{51, 12}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+	// Case: testUnits[1] RandomHermesFile{0, 50}
+	want, wantErr = "", fmt.Sprintf("The file ID provided %v wasn't in the required range [1,50]", testUnits[1].ID)
+	got, err = testUnits[1].FileName()
+	gotErr=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{0, 50}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{0, 50}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+	// Case: testUnits[2] RandomHermesFile{3, 100}
+	wantPrefix := "Hermes_03_"
+	got, err = testUnits[2].FileName()
+	if err!=nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	gotPrefix := got[0:10]
+	if gotPrefix != wantPrefix {
+		t.Errorf("{3,100}.FileName() failed expected prefix %v got prefix %v", wantPrefix, gotPrefix)
+	} 
+	// Case: testUnits[3] RandomHermesFile{12, 100}
+	wantPrefix = "Hermes_12_"
+	got, err = testUnits[3].FileName()
+	if err!=nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	gotPrefix = got[0:10]
+	if gotPrefix != wantPrefix {
+		t.Errorf("{12,100}.FileName() failed expected prefix %v got prefix %v", wantPrefix, gotPrefix)
+	} 
+	// Case: testUnits[4] RandomHermesFile{3, 0}
+	want, wantErr = "", "The file size provided 0 is not a positive number as required"
+	got, err = testUnits[4].FileName()
+	gotErr=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{3, 0}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{3, 0}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+	// Case: testUnits[5] RandomHermesFile{3, 1001}
+	want, wantErr = "", "The file size provided 1001 bytes exceeded the limit 1000 bytes"
+	got, err = testUnits[5].FileName()
+	gotErr=err.Error() // error in string form
+	if want!=got {
+		t.Errorf("{3, 1001}.FileName() failed expected %v got %v", want, got)
+	}
+	if wantErr!=gotErr {
+		t.Errorf("{3, 1001}.FileName() gave wrong error expected %v got %v", wantErr, gotErr)
+	}
+}
+
+func TestChecksum (t *testing.T) {
+	f := RandomHermesFile{11, 100}
+	fTwo := RandomHermesFile{13, 1000}
+	fCopy := RandomHermesFile{11, 100}
+	checksum, err := f.CheckSum() 
+	if(err!=nil){
+		t.Error(err)
+	}
+	checksumTwo, err := fTwo.CheckSum() 
+	if(err!=nil){
+		t.Error(err)
+	}
+	checksumCopy, err := fCopy.CheckSum() 
+	if(err!=nil){
+		t.Error(err)
+	}
+	if fmt.Sprintf("%x",checksum) != fmt.Sprintf("%x",checksumCopy) { //comparing checksums converted to strings as a slice can only be compared to nil
+		t.Errorf("Checksum returned different values for the same RandomHermesFiles {%v, %v}", f.ID, f.Size)
+	}
```

Comment:
to make this check using table driven test, your test struct can have three `RandomHermesFile` fields one for the main file, another one for the following (and/or preceding file), and then one for the copy of the main file. You could also just store the actual, expected checksum of the main file as `want`.